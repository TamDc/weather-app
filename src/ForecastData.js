import React from 'react'
const { default: Item } = require("./Item")
const { default: LoadingList } = require("./LoadingList")

const ForecastData = ({isLoading, data: {city, forecast}}) => {
  return(
    <div className="mt-md-3">
      <div>
        <h2 style={{color:'#0055a5'}}>
          { isLoading ? <img src="loading3.svg" alt="loading"/> : city }
        </h2>
      </div>
      <div className="container">
        <div className="row">
          { isLoading
            ? <LoadingList />
            : forecast.map(item => <Item key={item.id} {...item}/>)
          }
        </div>
      </div>
    </div>
  )
}

export default ForecastData
