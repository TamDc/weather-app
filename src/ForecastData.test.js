import React from 'react';
import renderer from 'react-test-renderer';
import ForecastData from './ForecastData';

const mock = {
  city: "San Joe",
  forecast: [
    {
      id: 1, 
      applicable_date: '2020-12-11', 
      min_temp: 20, 
      max_temp: 30
    },
    {
      id: 2, 
      applicable_date: '2020-12-12', 
      min_temp: 20, 
      max_temp: 30
    },
    {
      id: 3, 
      applicable_date: '2020-12-13', 
      min_temp: 20, 
      max_temp: 30
    },
    {
      id: 4, 
      applicable_date: '2020-12-14', 
      min_temp: 20, 
      max_temp: 30
    },
    {
      id: 5, 
      applicable_date: '2020-12-15', 
      min_temp: 20, 
      max_temp: 30
    }
  ]
}

test('renders loading phase', () => {
  const tree = renderer
    .create(<ForecastData isLoading data={mock}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders with data', () => {
  const tree = renderer
    .create(<ForecastData isLoading={false} data={mock}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
