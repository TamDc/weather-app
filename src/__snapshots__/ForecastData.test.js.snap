// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading phase 1`] = `
<div
  className="mt-md-3"
>
  <div>
    <h2
      style={
        Object {
          "color": "#0055a5",
        }
      }
    >
      <img
        alt="loading"
        src="loading3.svg"
      />
    </h2>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-2"
        style={
          Object {
            "maxWidth": "160px",
            "minWidth": "132px",
          }
        }
      >
        <div
          className="p-3 border"
          style={
            Object {
              "minHeight": "120px",
              "textAlign": "center",
            }
          }
        >
          <img
            alt="loading"
            className="mt-4"
            src="loading3.svg"
          />
        </div>
      </div>
      <div
        className="col-2"
        style={
          Object {
            "maxWidth": "160px",
            "minWidth": "132px",
          }
        }
      >
        <div
          className="p-3 border"
          style={
            Object {
              "minHeight": "120px",
              "textAlign": "center",
            }
          }
        >
          <img
            alt="loading"
            className="mt-4"
            src="loading3.svg"
          />
        </div>
      </div>
      <div
        className="col-2"
        style={
          Object {
            "maxWidth": "160px",
            "minWidth": "132px",
          }
        }
      >
        <div
          className="p-3 border"
          style={
            Object {
              "minHeight": "120px",
              "textAlign": "center",
            }
          }
        >
          <img
            alt="loading"
            className="mt-4"
            src="loading3.svg"
          />
        </div>
      </div>
      <div
        className="col-2"
        style={
          Object {
            "maxWidth": "160px",
            "minWidth": "132px",
          }
        }
      >
        <div
          className="p-3 border"
          style={
            Object {
              "minHeight": "120px",
              "textAlign": "center",
            }
          }
        >
          <img
            alt="loading"
            className="mt-4"
            src="loading3.svg"
          />
        </div>
      </div>
      <div
        className="col-2"
        style={
          Object {
            "maxWidth": "160px",
            "minWidth": "132px",
          }
        }
      >
        <div
          className="p-3 border"
          style={
            Object {
              "minHeight": "120px",
              "textAlign": "center",
            }
          }
        >
          <img
            alt="loading"
            className="mt-4"
            src="loading3.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders with data 1`] = `
<div
  className="mt-md-3"
>
  <div>
    <h2
      style={
        Object {
          "color": "#0055a5",
        }
      }
    >
      San Joe
    </h2>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-2"
        style={
          Object {
            "maxWidth": "160px",
            "minWidth": "132px",
          }
        }
      >
        <div
          className="p-3"
          style={
            Object {
              "border": "1px solid #45aee7",
              "minWidth": "inherit",
              "textAlign": "center",
            }
          }
        >
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "1.3rem",
              }
            }
          >
            Tomorrow
          </span>
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "0.9rem",
                "marginTop": "5px",
              }
            }
          >
            Min: 20°C
          </span>
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "0.9rem",
                "marginTop": "5px",
              }
            }
          >
            Max: 30°C
          </span>
        </div>
      </div>
      <div
        className="col-2"
        style={
          Object {
            "maxWidth": "160px",
            "minWidth": "132px",
          }
        }
      >
        <div
          className="p-3"
          style={
            Object {
              "border": "1px solid #45aee7",
              "minWidth": "inherit",
              "textAlign": "center",
            }
          }
        >
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "1.3rem",
              }
            }
          >
            Saturday
          </span>
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "0.9rem",
                "marginTop": "5px",
              }
            }
          >
            Min: 20°C
          </span>
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "0.9rem",
                "marginTop": "5px",
              }
            }
          >
            Max: 30°C
          </span>
        </div>
      </div>
      <div
        className="col-2"
        style={
          Object {
            "maxWidth": "160px",
            "minWidth": "132px",
          }
        }
      >
        <div
          className="p-3"
          style={
            Object {
              "border": "1px solid #45aee7",
              "minWidth": "inherit",
              "textAlign": "center",
            }
          }
        >
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "1.3rem",
              }
            }
          >
            Sunday
          </span>
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "0.9rem",
                "marginTop": "5px",
              }
            }
          >
            Min: 20°C
          </span>
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "0.9rem",
                "marginTop": "5px",
              }
            }
          >
            Max: 30°C
          </span>
        </div>
      </div>
      <div
        className="col-2"
        style={
          Object {
            "maxWidth": "160px",
            "minWidth": "132px",
          }
        }
      >
        <div
          className="p-3"
          style={
            Object {
              "border": "1px solid #45aee7",
              "minWidth": "inherit",
              "textAlign": "center",
            }
          }
        >
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "1.3rem",
              }
            }
          >
            Monday
          </span>
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "0.9rem",
                "marginTop": "5px",
              }
            }
          >
            Min: 20°C
          </span>
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "0.9rem",
                "marginTop": "5px",
              }
            }
          >
            Max: 30°C
          </span>
        </div>
      </div>
      <div
        className="col-2"
        style={
          Object {
            "maxWidth": "160px",
            "minWidth": "132px",
          }
        }
      >
        <div
          className="p-3"
          style={
            Object {
              "border": "1px solid #45aee7",
              "minWidth": "inherit",
              "textAlign": "center",
            }
          }
        >
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "1.3rem",
              }
            }
          >
            Tuesday
          </span>
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "0.9rem",
                "marginTop": "5px",
              }
            }
          >
            Min: 20°C
          </span>
          <span
            style={
              Object {
                "display": "block",
                "fontSize": "0.9rem",
                "marginTop": "5px",
              }
            }
          >
            Max: 30°C
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
